### ISSUE TITLE
Custom data generator ignored?

### ISSUE THREAD

**Issue by OmanF** [fscheck/FsCheck#506]
Hi.
Some metadata: `.Net Core Preview 8`; running via Rider's `xUnit.Net` GUI runner.
I'm came up with the following code (that I asked around quite a lot, seems OK F# code):
```fsharp
module GildedRoseTests

// Setup /////////////////////////////////////////////////////////////
open FsCheck
open FsCheck.Xunit
open Item

type NonLegendaryItem = NonLegendaryItem of Item
type LegendaryItem = LegendaryItem of Item

let genString15 = Gen.sized (fun s -> Gen.resize (min s 15) Arb.generate<NonNull<string>>)
                  |> Gen.map (fun (NonNull s) -> s)

let genNonNegativeOf f = gen {
    let! (NonNegativeInt n) = Arb.generate<NonNegativeInt>
    return (f n)
}

let genNonLegendaryItem = gen {
    let! name = Gen.map Name genString15
    let! quality = genNonNegativeOf Quality
    let! shelflife = genNonNegativeOf Days
    let! style = Gen.elements [Plain; Conjured]
    return {
        Name = name
        Quality = quality
        ShelfLife = shelflife
        Style = style
    }
}

let genLegendaryItem = genNonLegendaryItem |> Gen.map (fun item -> {item with Style = Legendary})

type Generator =
    static member NonLegendaryItem() =
        Gen.map NonLegendaryItem genNonLegendaryItem

    static member LegendaryItem() =
        Gen.map LegendaryItem genLegendaryItem

Arb.register<Generator>() |> ignore

// Test //////////////////////////////////////////////////////////////
[<Property>]
let ``This is a test`` (nonLegendaryItem: NonLegendaryItem) =
    match nonLegendaryItem with
    | NonLegendaryItem item ->
        let nli = repeat 100 decreaseQuality item
        nli.Quality = Quality 0
```

However, when clicking the Rider icon to run the test, the test fails with the following details:
![what_is_wrong](https://user-images.githubusercontent.com/7848797/64338257-9cfe1800-cfe1-11e9-905b-ea6bb46edff0.png)

As can be seen in the screenshot, while I specify a `NonLegendaryItem` as my parameter, the `Style` generated is `Legendary` after all.
Also, the `Quality` which is supposed to be derived as a `NonNegativeInt` is -1.

So, basically, my custom generator is ignored.

Is this a real issue or am I doing something wrong?

Thanks.

**Comment by OmanF** [comment]
Ok, I don't know whether it is my mistakes or something broken, but, I tried another approach I was suggested, using `Arb` instances (see attached screenshot), and again, got `Quality -1` when I specifically constructed the arbitrary type with `PositiveInt` (i.e. I went stricter than `NonNegativeInt`, in this implementation I barred 0 as well!)

BTW, `FsCheck` (and `FsCheck.Xunit`) at 2.14.0

![what_is_wrong](https://user-images.githubusercontent.com/7848797/64350361-cfffd600-cff8-11e9-916b-ec8b46df4dc8.png)

**Comment by kurtschelfthout** [comment]
Try passing the Generator type to the ProoertyAttribute, like in the example https://fscheck.github.io/FsCheck/RunningTests.html#Using-FsCheck-Xunit

**Comment by OmanF** [comment]
Still not...

![still_not](https://user-images.githubusercontent.com/7848797/64356504-473a6780-d003-11e9-9801-70967dcab5f2.png)

Tried all the permutations:

- Comment out the `Arb.register...` (since the example code doesn't have it)
- Not put the test code in its own module
- Put test code in its own module **but not** putting `Properties...` above
- Only putting `Properties...` above the module.
- Only putting `Property...` above the actual test code

In all those cases it's the same compile error: no instance of type `Generator`.

**Comment by OmanF** [comment]
Last comment is my mistake... I forgot to make an `Arbitraty` instance of my static members using `Arb.fromGen`.

I now can run the tests.

Thanks!
